runtime/utils/common/src/aws/smithy/kotlin/runtime/util/text/Text.kt

```diff
@@ -99,11 +99,20 @@ private const val upperHex: String = "0123456789ABCDEF"
 private val upperHexSet = upperHex.toSet()
 
 // $2.1 Percent-Encoding
-private fun Byte.percentEncode(): String = buildString(3) {
+@InternalApi
+fun Byte.percentEncodeTo(out: Appendable) {
     val code = toInt() and 0xff
-    append('%')
-    append(upperHex[code shr 4])
-    append(upperHex[code and 0x0f])
+    out.append('%')
+    out.append(upperHex[code shr 4])
+    out.append(upperHex[code and 0x0f])
+}
+
+@InternalApi
+fun Byte.percentEncode(): String {
+    val byte = this
+    return StringBuilder(3)
+        .apply { byte.percentEncodeTo(this) }
+        .toString()
```

Comment:
**Style**: The `apply` call requires aliasing `this` whereas `also` captures the context object as `it` (vs `this`) and can thus simplify the call chain:

```kotlin
fun Byte.percentEncode(): String = StringBuilder(3).also(::percentEncodeTo).toString()
```

Reply:
Yeah that's better, what Ian said.